### PR TITLE
[crail] Add support for Apache Crail (incubating)

### PR DESCRIPTION
**Pull request by patrickstuedi**

### CODE REVIEW

File: crail/src/main/java/com/yahoo/ycsb/db/CrailClient.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2015 YCSB contributors. All rights reserved.
```
Comment (busbey) [inline crail/src/main/java/com/yahoo/ycsb/db/CrailClient.java]: copyright year needs update

File: crail/src/main/java/com/yahoo/ycsb/db/package-info.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+/**
+ * YCSB binding for <a href="http://www.crail.io/">Crail</a>.
+ */
+package com.yahoo.ycsb.db;
```
Comment (busbey) [inline crail/src/main/java/com/yahoo/ycsb/db/package-info.java]: please put everything in a package specific to the db implementation. e.g. `com.yahoo.ycsb.db.crail`

File: crail/src/main/java/com/yahoo/ycsb/db/CrailClient.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    try {
+      CrailConfiguration crailConf = new CrailConfiguration();
+      this.client = CrailStore.newInstance(crailConf);
+      this.usertable = "usertable";
+      if (client.lookup(usertable).get() == null) {
+        client.create(usertable, CrailNodeType.TABLE, CrailStorageClass.DEFAULT, 
+          CrailLocationClass.DEFAULT, true).get().syncDir();
+      }
+      this.startTime = System.nanoTime();
+      System.out.println("YCSB/Crail client initialized");
```
Comment (busbey) [inline crail/src/main/java/com/yahoo/ycsb/db/CrailClient.java]: please use a logging framework instead of System.out. Preferably slf4j.

Comment (patrickstuedi) [reply crail/src/main/java/com/yahoo/ycsb/db/CrailClient.java]: Thanks for the comments Sean, sorry this is still due, I have time to work on this next week.

**Comment by busbey** [comment]
still working on this @patrickstuedi ?

**Comment by patrickstuedi** [comment]
@busbey yes. we have been working on a crail binary release which will be available at the official apache maven repository. That's required for this PR to build without the need to build crail. I hope this will be done in a week or so.. sorry for the delay!

**Comment by patrickstuedi** [comment]
I'll close this for the moment as start a new clean PR once the Crail binaries are in maven central.

**Comment by busbey** [comment]
Sounds good. Thanks!

**Pull request closed without merging**